### PR TITLE
Correctly set update config in disable_parameters

### PR DESCRIPTION
**Pull request by dafeda**
## Pre review checklist

- [ ] Read through the code changes carefully after finishing work
- [ ] Make sure tests pass locally (after every commit!)
- [ ] Prepare changes in small commits for more convenient review (optional)
- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Updated documentation
- [ ] Ensured that unit tests are added for all new behavior (See 
    [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)),
    and changes to existing code have good test coverage.

## Pre merge checklist
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: src/ert/analysis/configuration.py [hunk 1]
```diff
@@ -68,8 +74,14 @@ def observation_config(self) -> List[Tuple[str, Optional[List[int]]]]:
         ]
 
 
+if TYPE_CHECKING:
+    ConstrainedUpdateStepList = List[UpdateStep]
+else:
+    ConstrainedUpdateStepList = conlist(UpdateStep, min_items=1)
+
+
```
Comment (frode-aarstad) [inline src/ert/analysis/configuration.py:82]: Do we need two if these blocks in the same file ?

Comment (dafeda) [reply src/ert/analysis/configuration.py:82]: I don't see how I can do it with just one. Do you know of a way?

Comment (frode-aarstad) [reply src/ert/analysis/configuration.py:82]: Is it not similar to ie:

https://github.com/equinor/ert/blob/589da5f5751d9a51d77deb664c734f5edcf92dec/src/ert/analysis/_es_update.py#L35

Where you define typing on the top of the file or are you trying to do something else?

Comment (dafeda) [reply src/ert/analysis/configuration.py:82]: It is not the same as I am here defining both the type for static type-checking and a `conlist` for pydantic.
If it was just for type-checking, then I could get away with just one if.

**Review by frode-aarstad — APPROVED** [review]

**Comment by frode-aarstad** [comment]
LGTM. When you fix the tests

**Comment by dafeda** [comment]
Mind taking a look at this one @oyvindeide ?
Especially my claim that the test is safe to remove, and perhaps also comment on the getter and setter of disable parameters being typed differently.

**Comment by oyvindeide** [comment]
> Mind taking a look at this one @oyvindeide ?
> Especially my claim that the test is safe to remove, and perhaps also comment on the getter and setter of disable parameters being typed differently.

Why do you want to remove the test? The changes to the code seem unrelated to it.

**Comment by dafeda** [comment]
I wrote this in the commit message:

Remove test_parse_comma_list:
Because of mocking, this does not call the setter of EnKFMain
that sets update_configuration and will fail if the setter
in disable_parameters.py is called correctly with a list of
UpdateStep instead of UpdateConfiguration.

So, from my understanding the test fails when the setter is called correctly which is not great.

File: tests/unit_tests/all/plugins/test_parameter_disable.py [hunk 2]
```diff
@@ -10,18 +8,6 @@
...
-@pytest.mark.parametrize(
-    "input_string, expected", [("a", ["b", "c"]), ("a,b", ["c"]), ("a, b", ["c"])]
-)
-def test_parse_comma_list(tmpdir, monkeypatch, input_string, expected):
-    ert_mock = MagicMock()
-    ert_mock._observation_keys = ["OBSERVATION"]
-    ert_mock._parameter_keys = ["a", "b", "c"]
-
-    DisableParametersUpdate(ert_mock, storage=None).run(input_string)
-    assert ert_mock.update_configuration[0]["parameters"] == expected
```
Comment (oyvindeide) [inline tests/unit_tests/all/plugins/test_parameter_disable.py:22]:
```suggestion
    assert [
        param.name for param in ert_mock.update_configuration[0].parameters
    ] == expected
```

**Review by oyvindeide — COMMENTED** [review]
Forgot to read the commit-body 🙈 Suggest changing the test assert, as the test in itself seems to test something unreleated to the code changes.

File: tests/unit_tests/all/plugins/test_parameter_disable.py (hunk 2 above)
Comment (dafeda) [reply tests/unit_tests/all/plugins/test_parameter_disable.py:22]: That makes sense, thanks.
I've pushed a fix.

**Review by oyvindeide — APPROVED** [review]

**Pull request merged**